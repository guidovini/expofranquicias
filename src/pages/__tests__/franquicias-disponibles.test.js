import React from 'react'
import { render } from '@testing-library/react'
import { StaticQuery, useStaticQuery } from 'gatsby' // mocked

import FranquiciasDisponibles from '../franquicias-disponibles'

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      site: {
        siteMetadata: {
          title: `Default Starter`,
        },
      },
      allImageSharp: {
        edges: [
          {
            node: {
              fluid: {
                base64: `pretend-i-am-a-base64-encoded-image`,
                aspectRatio: 1,
                src: 'this is the src',
                srcSet: `asdfasdf`,
                sizes: `100 200 300`,
              },
            },
          },
        ],
      },
      allGoogleSheetListadoFranquiciasRow: {
        nodes: [
          {
            id: '1e773f6c-4336-5af2-a42c-b7f51c11affb',
            nombre: 'Autoseguro',
            categoria:
              'Comercializacion de Productos;Locales Comerciales;Productos de Venta',
            concepto: 'Comercialización de Purificadores de Agua',
            aniosOperacion: 37,
            numeroLocales: null,
            tamanoLocal: '6 - 16 m2',
            disponibilidadTerritorial: 'Todo el Ecuador',
            paisOrigen: 'Estados Unidos',
            tiempoRetornoInversion: '< 6 meses',
            inversion: 27500,
          },
        ],
      },
    })
  )
    .mockImplementationOnce(({ render }) =>
      render({
        allImageSharp: {
          edges: [
            {
              node: {
                fluid: {
                  base64: `pretend-i-am-a-base64-encoded-image`,
                  aspectRatio: 1,
                  src: 'this is the src',
                  srcSet: `asdfasdf`,
                  sizes: `100 200 300`,
                },
              },
            },
          ],
        },
        allGoogleSheetListadoFranquiciasRow: {
          nodes: [
            {
              id: '1e773f6c-4336-5af2-a42c-b7f51c11affb',
              nombre: 'Autoseguro',
              categoria:
                'Comercializacion de Productos;Locales Comerciales;Productos de Venta',
              concepto: 'Comercialización de Purificadores de Agua',
              aniosOperacion: 37,
              numeroLocales: null,
              tamanoLocal: '6 - 16 m2',
              disponibilidadTerritorial: 'Todo el Ecuador',
              paisOrigen: 'Estados Unidos',
              tiempoRetornoInversion: '< 6 meses',
              inversion: 27500,
            },
          ],
        },
      })
    )
    .mockImplementationOnce(({ render }) =>
      render({
        allImageSharp: {
          edges: [
            {
              node: {
                fluid: {
                  base64: `pretend-i-am-a-base64-encoded-image`,
                  aspectRatio: 1,
                  src: 'this is the src',
                  srcSet: `asdfasdf`,
                  sizes: `100 200 300`,
                },
              },
            },
          ],
        },
      })
    )
  useStaticQuery.mockReturnValueOnce({
    site: {
      siteMetadata: {
        title: `Default Starter`,
        description: `Test description`,
        author: `Test author`,
      },
    },
  })
})

describe('Franquicias Disponibles Page', () => {
  test('Should_RenderFranquiciasDisponiblesPageCorrectly', () => {
    const { context } = render(<FranquiciasDisponibles />)
    expect(context).toMatchSnapshot()
  })

  test('Should_RenderFranquiciasDisponiblesList', () => {
    const { getByTestId } = render(<FranquiciasDisponibles />)
    expect(getByTestId('franchise-list')).toBeInTheDocument()
  })
})
