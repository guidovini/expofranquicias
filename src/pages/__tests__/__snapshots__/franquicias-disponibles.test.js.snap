// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Franquicias Disponibles Page Should_RenderFranquiciasDisponiblesPageCorrectly 1`] = `undefined`;
